fili-core/src/main/java/com/yahoo/bard/webservice/web/endpoints/DimensionSearchServlet.java

```diff
@@ -0,0 +1,198 @@
+
+    private final DimensionDictionary dimensionDictionary;
+    private final RequestMapper requestMapper;
+    private final ResponseFormatResolver formatResolver;
+    private final MetadataExceptionHandler exceptionHandler;
+
+    /**
+     * Constructor.
+     *
+     * @param dimensionDictionary  All dimensions
+     * @param logicalTableDictionary  All logical tables
+     * @param requestMapper  Mapper to change the API request if needed
+     * @param objectMappers  JSON tools
+     * @param formatResolver  The formatResolver for determining correct response format
+     * @param exceptionHandler  Injection point for handling response exceptions
+     */
+    @Inject
+    public DimensionSearchServlet(
+            DimensionDictionary dimensionDictionary,
+            LogicalTableDictionary logicalTableDictionary,
+            @Named(DimensionsApiRequest.REQUEST_MAPPER_NAMESPACE) RequestMapper requestMapper,
+            ObjectMappersSuite objectMappers,
+            ResponseFormatResolver formatResolver,
+            @Named(DimensionsApiRequest.EXCEPTION_HANDLER_NAMESPACE) MetadataExceptionHandler exceptionHandler
+    ) {
+        super(objectMappers);
+        this.dimensionDictionary = dimensionDictionary;
+        this.requestMapper = requestMapper;
+        this.formatResolver = formatResolver;
+        this.exceptionHandler = exceptionHandler;
+    }
+
+    /**
+     * Search endpoint. Handles text search queries against a single dimension.
+     *
+     * @param dimensionName  The dimension
+     * @param searchQuery The query to search the dimensions with
+     * @param perPage  The number of rows per page
+     * @param page  The page number
+     * @param format  The format of the response
+     * @param downloadFilename If present, indicates the response should be downloaded by the client with the provided
+     * filename. Otherwise indicates the response should be rendered in the browser.
+     * @param uriInfo The injected UriInfo
+     * @param containerRequestContext The injected request context
+     *
+     * @return A paginated list of dimensions and their metadata determined to best match the search query.
+     */
+    @GET
+    @Timed
+    public Response searchDimension(
+            @PathParam("dimensionName") String dimensionName,
+            @DefaultValue("") @NotNull @QueryParam("perPage") String perPage,
+            @DefaultValue("") @NotNull @QueryParam("page") String page,
+            @QueryParam("format") String format,
+            @QueryParam("filename") String downloadFilename,
+            @QueryParam("query") String searchQuery,
+            @Context final UriInfo uriInfo,
+            @Context final ContainerRequestContext containerRequestContext
+    ) {
+        DimensionsApiRequest apiRequest = null;
+        try {
+            RequestLog.startTiming(this);
+            RequestLog.record(new DimensionRequest(dimensionName, "yes"));
```

Comment:
this is odd.  I should follow up in the original code whether this is doing the right thing.  Don't worry about it.  I'm making a note more for myself here.